### PR TITLE
Refactor usesCalories in EnergyFormatter

### PR DESCRIPTION
**Pull request by mangerlahn**
This pull request refactors the TODO in Energy Formatter, as Locale.regionCode is now implemented.

### CODE REVIEW

**Comment by mangerlahn** [comment]
@swift-ci please smoke test

**Comment by pushkarnk** [comment]
@swift-ci please test

File: Foundation/EnergyFormatter.swift [hunk 1]
```diff
@@ -177,25 +177,14 @@ open class EnergyFormatter: Formatter {
...
-        default: return false
-        }
+    
+    /// Regions that use calories
+    private static let caloriesRegions: Set<String> = ["GB", "US"]
+    
+    /// Whether the region uses calories
+    private var usesCalories: Bool {
+        guard let region = numberFormatter.locale.regionCode else { return false }
+        return EnergyFormatter.caloriesRegions.contains(region)
```
Comment (millenomi) [inline Foundation/EnergyFormatter.swift]: This isn't strictly equivalent — regions that aren't `en`, `haw`, `kw`, `cy` or `gv` in language will get calories where they don't elsewhere.

Comment (mangerlahn) [reply Foundation/EnergyFormatter.swift]: I see. So I would go back to Locale.identifier there?

**Comment by spevans** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Pull request merged**